### PR TITLE
Ui

### PR DESCRIPTION
**Pull request by davidabrahams**

### CODE REVIEW

**Comment by davidabrahams** [comment]
The `ui` branch now has my simplistic `ui` with @nmohamed 's voice recognition code integrated.

File: app/src/main/java/com/mobileproto/hireddit/hireddit/MainActivity.java [hunk 1]
```diff
@@ -1,41 +1,34 @@
...
+    public static final int HOME_SCREEN_CLICK = 1;
+
     @Override
     protected void onCreate(Bundle savedInstanceState)
     {
         super.onCreate(savedInstanceState);
+        manager = getSupportFragmentManager();
         setContentView(R.layout.activity_main);
+        switchFragment(MainFragment.newInstance());
         Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
```
Comment (nmohamed) [inline app/src/main/java/com/mobileproto/hireddit/hireddit/MainActivity.java:28]: butterknife?

File: app/src/main/java/com/mobileproto/hireddit/hireddit/MainActivity.java [hunk 2]
```diff
@@ -62,5 +55,33 @@ public boolean onOptionsItemSelected(MenuItem item)
...
     }
 
+    private void switchFragment(Fragment f)
+    {
+        FragmentTransaction transaction = manager.beginTransaction();
+        transaction.replace(R.id.container, f).addToBackStack(null);
+        transaction.commit();
+    }
+
+    private void switchFragment(Fragment f, int customAnimationIn, int customAnimationOut)
```
Comment (nmohamed) [inline app/src/main/java/com/mobileproto/hireddit/hireddit/MainActivity.java:68]: try not to name both switchfragment (maybe switchFragmentAnimation or something)

File: app/src/main/java/com/mobileproto/hireddit/hireddit/MainFragment.java [hunk 3]
```diff
@@ -1,29 +1,108 @@
...
+
+    @Bind(R.id.alienImage) ImageView alienImage;
+
+    /**
+     * Use this factory method to create a new instance of
+     * this fragment using the provided parameters.
+     *
+     * @return A new instance of fragment MainFragment.
+     */
+    public static MainFragment newInstance()
```
Comment (nmohamed) [inline app/src/main/java/com/mobileproto/hireddit/hireddit/MainFragment.java:36]: is it necessary to name this mainfragment? i also just don't know what a newInsistance is

**Comment by nmohamed** [comment]
other than that, seems good :peach: 
*note: theres a small problem in the speech part. if you press the button, don't say something/what you say isn't recognized, you'll have to press the button one more time (which gives error) then another time to start listening again. it's minor + pretty simple to solve. also if you say gibberish and it doesnt recognize what you say you get an error everytime partial results tries to get partial results

**Comment by davidabrahams** [comment]
@nmohamed could you make a new branch off master

```
git checkout master
git checkout -b speech-bug-fixes
```

and then submit a pull request with the fixes?

File: app/src/main/java/com/mobileproto/hireddit/hireddit/MainActivity.java (hunk 1 above)
Comment (davidabrahams) [inline app/src/main/java/com/mobileproto/hireddit/hireddit/MainActivity.java:28]: you can't really butterknife this line. `MainFragment.newInstance()` is very similar to

```
new MainFragment()
```

File: app/src/main/java/com/mobileproto/hireddit/hireddit/MainActivity.java (hunk 2 above)
Comment (davidabrahams) [inline app/src/main/java/com/mobileproto/hireddit/hireddit/MainActivity.java:68]: I kinda named them the same thing on purpose to make it clear that both functions do the exact same thing, just one of them allows you to pick the animation the `Activity` uses

File: app/src/main/java/com/mobileproto/hireddit/hireddit/MainFragment.java (hunk 3 above)
Comment (davidabrahams) [inline app/src/main/java/com/mobileproto/hireddit/hireddit/MainFragment.java:36]: `newInstance` is a function that returns a newly created `MainFragment` object. What do you mean is it necessary to name this mainfragment?

**Comment by davidabrahams** [comment]
@YuzhongHuang @LucyWilcox could one of you review this pull request so that we could get it merged?

File: app/src/main/java/com/mobileproto/hireddit/hireddit/MainActivity.java (hunk 2 above)
Comment (LucyWilcox) [inline app/src/main/java/com/mobileproto/hireddit/hireddit/MainActivity.java:68]: Maybe add a comment explaining that in the code

**Comment by LucyWilcox** [comment]
:dart:

File: app/src/main/java/com/mobileproto/hireddit/hireddit/MainActivity.java (hunk 2 above)
Comment (davidabrahams) [inline app/src/main/java/com/mobileproto/hireddit/hireddit/MainActivity.java:68]: alright there's a comment. Are we good to merge?

File: app/src/main/java/com/mobileproto/hireddit/hireddit/MainActivity.java (hunk 1 above)
Comment (nmohamed) [inline app/src/main/java/com/mobileproto/hireddit/hireddit/MainActivity.java:28]: I meant butterknifing the toolbar not the view

**Pull request merged**